### PR TITLE
spanner: add BatchQuery sample

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

File: spanner/spanner_snippets/snippet.go [hunk 1]
```diff
@@ -485,6 +486,53 @@ func readStaleData(ctx context.Context, w io.Writer, client *spanner.Client) err
...
+			log.Printf("Partition (%d) %v", i, s)
+			recordCount++
+		}
+	}
+	log.Printf("Total partition count: %v", len(partitions))
+	log.Printf("Total record count: %v", recordCount)
+	return nil
+}
+
+// [START spanner_batch_client]
```
Comment (broady) [inline spanner/spanner_snippets/snippet.go]: END

File: spanner/spanner_snippets/snippet.go [hunk 2]
```diff
@@ -485,6 +486,53 @@ func readStaleData(ctx context.Context, w io.Writer, client *spanner.Client) err
...
+			var s Singer
+			if err := row.ToStruct(&s); err != nil {
+				return err
+			}
+			log.Printf("Partition (%d) %v", i, s)
+			recordCount++
+		}
+	}
+	log.Printf("Total partition count: %v", len(partitions))
+	log.Printf("Total record count: %v", recordCount)
```
Comment (broady) [inline spanner/spanner_snippets/snippet.go]: `fmt.Fprintf(w, ...)` everywhere.

**Comment by broady** [comment]
R= @tbpg

File: spanner/spanner_snippets/snippet.go [hunk 3]
```diff
@@ -485,6 +486,53 @@ func readStaleData(ctx context.Context, w io.Writer, client *spanner.Client) err
...
+// [START spanner_batch_client]
+
+func readBatchData(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	txn, err := client.BatchReadOnlyTransaction(ctx, spanner.StrongRead())
+	if err != nil {
+		return err
+	}
+	defer txn.Close()
+
+	// Singer represents the elements in a row from the Singers table.
```
Comment (tbpg) [inline spanner/spanner_snippets/snippet.go]: Maybe simplify this to:

`// Singer represents a row in the Singers table.`

?

**Comment by broady** [comment]
Did you investigate this error at all?

```
=== RUN   TestSample
Usage of /tmp/go-build060357401/github.com/GoogleCloudPlatform/golang-samples/spanner/spanner_snippets/_test/spanner_snippets.test:
  -test.bench regexp
    	run only benchmarks matching regexp
  -test.benchmem
    	print memory allocations for benchmarks
  -test.benchtime d
    	run each benchmark for duration d (default 1s)
  -test.blockprofile file
    	write a goroutine blocking profile to file
  -test.blockprofilerate rate
    	set blocking profile rate (see runtime.SetBlockProfileRate) (default 1)
  -test.count n
    	run tests and benchmarks n times (default 1)
  -test.coverprofile file
    	write a coverage profile to file
  -test.cpu list
    	comma-separated list of cpu counts to run each test with
  -test.cpuprofile file
    	write a cpu profile to file
  -test.memprofile file
    	write a memory profile to file
  -test.memprofilerate rate
    	set memory profiling rate (see runtime.MemProfileRate)
  -test.mutexprofile string
    	write a mutex contention profile to the named file after execution
  -test.mutexprofilefraction int
    	if >= 0, calls runtime.SetMutexProfileFraction() (default 1)
  -test.outputdir dir
    	write profiles to dir
  -test.parallel n
    	run at most n tests in parallel (default 2)
  -test.run regexp
    	run only tests and examples matching regexp
  -test.short
    	run smaller test suite to save time
  -test.timeout d
    	fail test binary execution after duration d (0 means unlimited)
  -test.trace file
    	write an execution trace to file
  -test.v
    	verbose: print additional output
FAIL	github.com/GoogleCloudPlatform/golang-samples/spanner/spanner_snippets	517.912s
```

**Pull request merged**